server/trillian_log_server.go

```diff
@@ -157,6 +181,20 @@ func (t *TrillianLogServer) commitAndLog(tx storage.LogTX, op string) error {
 	return err
 }
 
+func protoToLeaf(proto *trillian.LeafProto) trillian.LogLeaf {
+	return trillian.LogLeaf{Leaf: trillian.Leaf{LeafHash: proto.LeafHash, LeafValue: proto.LeafData, ExtraData: proto.ExtraData}}
+}
+
+func protosToLeaves(protos []*trillian.LeafProto) []trillian.LogLeaf {
+	leaves := make([]trillian.LogLeaf, 0, len(protos))
+
+	for _, proto := range protos {
```

Comment:
I've allocated the capacity at the start so the appends won't do allocation. I think I prefer this to leaving holes in the result. I think our APIs are inconsistent on this so we should maybe decide.